### PR TITLE
Max/site creation

### PR DESCRIPTION
**Pull request by maxwolffe**
Not ready for merge. Just to open up discussion, is there a better way to do this?

@charlesx2013 - I'm having trouble getting mini-site creation working here. 

I'm making a request to the minisite endpoint, and am sending the following Json:

{"mini_site":{"photos":[],"zip_code":92867,"description":"Woah, looks great!","street":"177","name":"MiniSiteTest","state":"CA","site_id":2,"longitude":"0","latitude":"0","city":"Orinda"}}

Any thoughts?

### CODE REVIEW

**Comment by nishdesai** [comment]
I think this is the best way to get code reviewed.

**Comment by negativetwelve** [comment]
you can put fancy :warning: tags around the top line saying something like "not ready to merge" :warning:

**Comment by maxwolffe** [comment]
@nishdesai , @charlesx2013 said this is working and ready to merge.

But ..er.... dat code ship. Idk.

**Comment by nishdesai** [comment]
@maxwolffe don't worry about codeship. It just doesn't have the files it needs to be able to test the build.

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/FieldReports/AddFieldReportFragment.java [hunk 1]
```diff
@@ -114,8 +114,10 @@ public void onResponse(FieldReport fieldReport) {
      * View.OnClickListener methods
      */
     public void onClick(View view) {
+        Log.e("Click", "Button");
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/FieldReports/AddFieldReportFragment.java]: we can remove this

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/FieldReports/AddFieldReportFragment.java [hunk 2]
```diff
@@ -114,8 +114,10 @@ public void onResponse(FieldReport fieldReport) {
...
         switch (view.getId()) {
             case R.id.take_photo_button:
+                Log.e("Click", "Take Photo");
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/FieldReports/AddFieldReportFragment.java]: and this

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Networking/MiniSites/MiniSiteSerializer.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        jgen.writeStringField("city", value.getCity());
+        jgen.writeStringField("state", value.getState());
+        jgen.writeNumberField("zip_code", value.getZipCode());
+        jgen.writeNumberField("latitude", value.getLatitude());
+        jgen.writeNumberField("longitude", value.getLongitude());
+        jgen.writeNumberField("site_id", value.getSiteId());
+        jgen.writeArrayFieldStart("photos_attributes");
+        for (Photo photo : value.getPhotos()) {
+            jgen.writeStartObject();
+//            jgen.writeNumberField("id", photo.getId());
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Networking/MiniSites/MiniSiteSerializer.java]: we can also delete this line

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Networking/BaseRequest.java [hunk 4]
```diff
@@ -30,8 +30,8 @@
     private Response.Listener listener;
     private Response.Listener errorListener;
 
-//    private static final String baseURL = "http://192.168.0.111:3000/api/v1/";
-    private static final String baseURL = "https://intense-reaches-1457.herokuapp.com/api/v1/";
+    private static final String baseURL = "http://192.168.0.111:3000/api/v1/";
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Networking/BaseRequest.java]: no

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/MiniSites/MiniSiteFragment.java [hunk 5]
```diff
@@ -99,7 +99,7 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
     public void onResume() {
         super.onResume();
         mParentActivity.setMenuAction(false);
-        getMiniSiteRequest(mMiniSite);
+//        getMiniSiteRequest(mMiniSite);
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/MiniSites/MiniSiteFragment.java]: Are we no longer updating when the app re-opens? That would make sense, and if it's the case, then we can just get rid of this line entirely.

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/MiniSites/CreateMiniSiteFragment.java [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+        }
+        else if (requestCode == SELECT_PHOTO_REQUEST && resultCode == MainActivity.RESULT_OK){
+            Uri targetUri = data.getData();
+            try {
+                Bitmap photo = BitmapFactory.decodeStream(mMainActivity.getContentResolver().openInputStream(targetUri));
+                miniSiteImageView.setImageBitmap(photo);
+            }
+            catch (FileNotFoundException e){
+                e.printStackTrace();
+            }
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/MiniSites/CreateMiniSiteFragment.java]: what would happen in the case that neither of these if-conditions are met?

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/MiniSites/CreateMiniSiteFragment.java [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        if (takePictureIntent.resolveActivity(mMainActivity.getPackageManager()) != null) {
+            File photoFile = null;
+            try {
+                photoFile = createImageFile();
+            } catch (IOException ex) {
+                Log.e("Mini Site Photo", "Error");
+            }
+            if (photoFile != null) {
+                //takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT,
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/MiniSites/CreateMiniSiteFragment.java]: can probably remove these too

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/MiniSites/CreateMiniSiteFragment.java [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+        mStateField = (EditText)view.findViewById(R.id.create_mini_site_state);
+        takePhoto.setOnClickListener(this);
+        selectPhoto.setOnClickListener(this);
+        submitButton.setOnClickListener(this);
+    }
+
+    public void createMiniSiteRequest(MiniSite miniSite){
+        HashMap<String, JSONObject> params = new HashMap<String, JSONObject>();
+
+        CreateMiniSiteRequest createMiniSiteRequest = new CreateMiniSiteRequest(getActivity(), miniSite, params, new Response.Listener<MiniSite>() {
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/MiniSites/CreateMiniSiteFragment.java]: split up arguments for readability

**Comment by nishdesai** [comment]
Looks good overall. As usual, a few stylistic changes her and there. Also url... -_- @charlesx2013 @maxwolffe

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/MiniSites/CreateMiniSiteFragment.java (hunk 6 above)
Comment (charlesx2013) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/MiniSites/CreateMiniSiteFragment.java]: nothing

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/MiniSites/MiniSiteFragment.java (hunk 5 above)
Comment (charlesx2013) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/MiniSites/MiniSiteFragment.java]: keep this here, we can talk about this later!

**Pull request merged**